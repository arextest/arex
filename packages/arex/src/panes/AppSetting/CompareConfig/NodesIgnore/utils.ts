import { DataNode } from 'antd/lib/tree';

export function getNodes(object: object, basePath = ''): DataNode[] {
  const entries = Object.entries(object);
  return entries.map(([key, value]) => {
    const losslessValue = value.isLosslessNumber ? value.value : value;
    const isSimpleArray =
      Array.isArray(losslessValue) && ['number', 'string'].includes(typeof losslessValue[0]);
    const isObject = typeof losslessValue === 'object';

    const path = basePath + key + '/';

    return losslessValue && isObject && !isSimpleArray
      ? {
          title: key,
          key: path,
          children: getNodes(
            Array.isArray(losslessValue) ? losslessValue[0] || {} : losslessValue,
            path,
          ),
        }
      : { title: key, key: path, value: losslessValue };
  });
}
