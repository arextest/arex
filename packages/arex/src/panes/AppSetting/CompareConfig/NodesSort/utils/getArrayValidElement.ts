/**
 * 获取一个数组中的有效元素（非数组类型的对象或者基本数据类型）
 * @param array
 */
export function getArrayValidElement(array: any[]): any {
  const sample = array[0];
  return Array.isArray(sample) ? getArrayValidElement(sample) : sample;
}
