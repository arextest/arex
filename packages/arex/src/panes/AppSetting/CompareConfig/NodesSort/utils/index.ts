export { getArrayNodes } from './getArrayNodes';
export { getArrayValidElement } from './getArrayValidElement';
export { getSortArray } from './getSortArray';
export { getSortNodes } from './getSortNodes';
