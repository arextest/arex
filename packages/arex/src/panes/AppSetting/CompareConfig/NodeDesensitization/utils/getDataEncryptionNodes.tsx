import { Badge } from 'antd';
import { DataNode } from 'antd/lib/tree';
import React from 'react';

import { QueryEncryptionNode } from '@/services/ComparisonService';

import { getArrayValidElement } from '../../NodesSort/utils/getArrayValidElement';

/**
 * 获取 object 的节点，并为已配置过的数组节点添加圆点提示
 * @param object 契约对象
 * @param basePath 节点路径
 * @param desensitizationNodeList 已配置的节点列表
 * @param color 圆点颜色
 */
export function getDataDesensitizationNodes(
  object: object,
  basePath = '',
  desensitizationNodeList?: QueryEncryptionNode[],
  color?: string,
): DataNode[] {
  const entries = Object.entries(object);
  return entries.map(([key, value]) => {
    const losslessValue = value.isLosslessNumber ? value.value : value;

    const path = basePath + '/' + key;
    return losslessValue && typeof losslessValue === 'object'
      ? {
          title: key,
          key: path,
          children: getDataDesensitizationNodes(
            Array.isArray(losslessValue)
              ? getArrayValidElement(losslessValue) || {}
              : losslessValue,
            path,
            desensitizationNodeList,
            color,
          ),
          icon: desensitizationNodeList?.find((node) => '/' + node.path.join('/') === path) && (
            <Badge color={color} /> // 已配置过的节点使用圆点进行提示
          ),
        }
      : {
          title: key,
          key: path,
          value: losslessValue,
          icon: desensitizationNodeList?.find((node) => '/' + node.path.join('/') === path) && (
            <Badge color={color} /> // 已配置过的节点使用圆点进行提示
          ),
        };
  });
}
