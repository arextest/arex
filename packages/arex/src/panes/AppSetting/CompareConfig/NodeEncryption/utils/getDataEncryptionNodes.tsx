import { Badge } from 'antd';
import { DataNode } from 'antd/lib/tree';
import React from 'react';

import { QueryEncryptionNode } from '@/services/ComparisonService';

import { getArrayValidElement } from '../../NodesSort/utils/getArrayValidElement';

/**
 * 获取 object 的节点，并为已配置过的数组节点添加圆点提示
 * @param object 契约对象
 * @param basePath 节点路径
 * @param encryptionNodeList 已配置的节点列表
 * @param color 圆点颜色
 */
export function getDataEncryptionNodes(
  object: object,
  basePath = '',
  encryptionNodeList?: QueryEncryptionNode[],
  color?: string,
): DataNode[] {
  const entries = Object.entries(object);
  return entries.map(([key, value]) => {
    const losslessValue = value.isLosslessNumber ? value.value : value;

    const path = basePath + '/' + key;
    return losslessValue && typeof losslessValue === 'object'
      ? {
          title: key,
          key: path,
          children: getDataEncryptionNodes(
            Array.isArray(losslessValue)
              ? getArrayValidElement(losslessValue) || {}
              : losslessValue,
            path,
            encryptionNodeList,
            color,
          ),
          icon: encryptionNodeList?.find((node) => '/' + node.path.join('/') === path) && (
            <Badge color={color} /> // 已配置过的节点使用圆点进行提示
          ),
        }
      : {
          title: key,
          key: path,
          value: losslessValue,
          icon: encryptionNodeList?.find((node) => '/' + node.path.join('/') === path) && (
            <Badge color={color} /> // 已配置过的节点使用圆点进行提示
          ),
        };
  });
}
