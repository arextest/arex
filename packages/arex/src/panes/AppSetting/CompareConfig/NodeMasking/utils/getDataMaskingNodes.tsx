import { Badge } from 'antd';
import { DataNode } from 'antd/lib/tree';
import React from 'react';

import { getArrayValidElement } from '@/panes/AppSetting/CompareConfig/NodesSort/utils/getArrayValidElement';
import { SortNode } from '@/services/ComparisonService';

/**
 * 获取 object 的节点，并为已配置过的数组节点添加圆点提示
 * @param object 契约对象
 * @param basePath 节点路径
 * @param sortNodeList 已配置的节点列表
 * @param color 圆点颜色
 */
export function getDataMaskingNodes(
  object: object,
  basePath = '',
  sortNodeList?: SortNode[],
  color?: string,
): DataNode[] {
  const entries = Object.entries(object);
  return (
    entries
      .map(([key, value]) => {
        const losslessValue = value.isLosslessNumber ? value.value : value;

        const path = basePath + key + '/';
        return losslessValue && typeof losslessValue === 'object'
          ? {
              title: key,
              key: path,
              children: getDataMaskingNodes(
                Array.isArray(losslessValue)
                  ? getArrayValidElement(losslessValue) || {}
                  : losslessValue,
                path,
                sortNodeList,
                color,
              ),
              icon: sortNodeList?.find((node) => node.path === path)?.pathKeyList?.length && (
                <Badge color={color} /> // 已配置过的节点使用圆点进行提示
              ),
            }
          : {
              title: key,
              key: path,
              value: losslessValue,
            };
      })
      // 过滤非数组子节点
      .filter((item) => item.children || Array.isArray(item.value))
  );
}
