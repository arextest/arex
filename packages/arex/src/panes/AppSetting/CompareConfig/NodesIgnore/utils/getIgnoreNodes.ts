import { DataNode } from 'antd/lib/tree';

import { getArrayValidElement } from '@/panes/AppSetting/CompareConfig/NodesSort/utils';

/**
 * 获取 object 的节点信息
 * @param object
 * @param basePath
 */
export function getIgnoreNodes(object: object, basePath = ''): DataNode[] {
  const entries = Object.entries(object);
  return entries.map(([key, value]) => {
    const losslessValue = value.isLosslessNumber ? value.value : value;
    const isSimpleArray =
      Array.isArray(losslessValue) && ['number', 'string'].includes(typeof losslessValue[0]);
    const isObject = typeof losslessValue === 'object';

    const path = basePath + key + '/';

    return losslessValue && isObject && !isSimpleArray
      ? {
          title: key,
          key: path,
          children: getIgnoreNodes(
            Array.isArray(losslessValue)
              ? getArrayValidElement(losslessValue) || {}
              : losslessValue,
            path,
          ),
        }
      : { title: key, key: path, value: losslessValue };
  });
}
