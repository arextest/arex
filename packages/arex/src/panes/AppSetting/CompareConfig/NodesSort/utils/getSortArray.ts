import { NodesSortProps } from '@/panes/AppSetting/CompareConfig/NodesSort';
import { getArrayValidElement } from '@/panes/AppSetting/CompareConfig/NodesSort/utils/getArrayValidElement';

/**
 * 根据 path获取 contract 的非数组元素节点
 * @param key
 * @param contract
 */
export function getSortArray(key: string, contract: NodesSortProps['contractParsed']) {
  let value: any = undefined;
  key
    .split('/')
    .filter(Boolean)
    .forEach((k, i) => {
      value =
        i === 0 ? contract[k] : Array.isArray(value) ? getArrayValidElement(value)?.[k] : value[k];
    });

  return value;
}
