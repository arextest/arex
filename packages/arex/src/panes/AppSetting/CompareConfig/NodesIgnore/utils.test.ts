import { describe, expect, test } from 'vitest';

import { getNodes } from './utils';
describe('convert sort tree', () => {
  test('simple object', () => {
    const objectData = { name: 'Helen', age: 20 };
    expect(getNodes(objectData)).toMatchSnapshot();
  });

  test('complex object', () => {
    const objectData = {
      name: 'Helen',
      age: 20,
      hobby: ['music', 'football'],
      assets: {
        house: 'shanghai',
        car: ['BMW', 'Benz'],
        savings: 1000000,
      },
    };
    expect(getNodes(objectData)).toMatchSnapshot();
  });
});
