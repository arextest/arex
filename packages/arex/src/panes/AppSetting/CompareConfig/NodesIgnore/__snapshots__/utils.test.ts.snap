// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`convert sort tree > complex object 1`] = `
[
  {
    "key": "name/",
    "title": "name",
    "value": "Helen",
  },
  {
    "key": "age/",
    "title": "age",
    "value": 20,
  },
  {
    "key": "hobby/",
    "title": "hobby",
    "value": [
      "music",
      "football",
    ],
  },
  {
    "children": [
      {
        "key": "assets/house/",
        "title": "house",
        "value": "shanghai",
      },
      {
        "key": "assets/car/",
        "title": "car",
        "value": [
          "BMW",
          "Benz",
        ],
      },
      {
        "key": "assets/savings/",
        "title": "savings",
        "value": 1000000,
      },
    ],
    "key": "assets/",
    "title": "assets",
  },
]
`;

exports[`convert sort tree > simple object 1`] = `
[
  {
    "key": "name/",
    "title": "name",
    "value": "Helen",
  },
  {
    "key": "age/",
    "title": "age",
    "value": 20,
  },
]
`;
